### ISSUE TITLE
Allow users to edit their email address and password via edit profile (or implement settings)

### ISSUE THREAD

**Issue by jesicarson** [participedia/frontend#870]
There was a UI designed for settings. I'll find and post here.

**Comment by jesicarson** [comment]
The old mockup below was based on what .net offers for user settings. I think change password, change email address and maybe (eventually?) set language would be useful to add to what we currently offer on xyz (name, location, bio). Don't think we need time zone. @plscully, @scottofletcher do you agree?

@dethe do you think we can just add these extras to the edit profile page or should there be a separate settings page for users? 

![artboard 87 copy 3-100](https://user-images.githubusercontent.com/13560530/42339468-e974a208-8041-11e8-9156-a0d7618c5240.jpg)

**Comment by plscully** [comment]
These settings work for me. And I like the “set language”  option, although
I assume Dethe will have ideas on whether it’s necessary and/or how to
implement it so that it works with localization.



- pat



*From:* jesicarson <notifications@github.com>
*Sent:* Thursday, July 5, 2018 2:06 PM
*To:* participedia/frontend <frontend@noreply.github.com>
*Cc:* Patrick Scully <pscully@clearviewconsultingllc.com>; Mention <
mention@noreply.github.com>
*Subject:* Re: [participedia/frontend] Allow users to edit their email
address and password via edit profile (or implement settings) (#870)



The old mockup below was based on what .net offers for user settings. I
think change password, change email address and maybe (eventually?) set
language would be useful to add to what we currently offer on xyz (name,
location, bio). Don't think we need time zone. @plscully
<https://github.com/plscully>, @scottofletcher
<https://github.com/scottofletcher> do you agree?

@dethe <https://github.com/dethe> do you think we can just add these extras
to the edit profile page or should there be a separate settings page for
users?

[image: artboard 87 copy 3-100]
<https://user-images.githubusercontent.com/13560530/42339468-e974a208-8041-11e8-9156-a0d7618c5240.jpg>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/participedia/frontend/issues/870#issuecomment-402806684>,
or mute the thread

**Comment by scottofletcher** [comment]
looks good! I was just on the Canada Revenue Agency's website this morning and they had the same "punctuation not allowed except for..." except they had examples in brackets as well - might we do this also? It would just read "punctuation not allowed except for periods (.), hyphens (-), apostrophes ('), and underscores (_)."

**Comment by jesicarson** [comment]
Thank you.
Scott the mockup shown above is pretty old, including the explanatory text. So the whole page should be reviewed, and sure if you want examples we can do that. Should update the [copy deck](https://docs.google.com/document/d/1eUVXSBHN_QqR5O_MCNHpiplAAr51K3EM5yOo8w05I5k/edit) as we do this.
This also reminds me that "username" and "name" might be 2 different things on xyz, as I can't sign in with the name entered on my profile page at the moment. This is confusing and we should sort it out. Will bring up with Dethe at our working session today. https://github.com/participedia/frontend/issues/873

**Comment by jesicarson** [comment]
@dethe: 
Add Auth0 powered fields for password and email, along with a terms of use check box linking to our TOU page, to edit profile page. 
xyz no longer supports usernames (remove "username/" from auth0 login window). only email is used to login. multiple email addresses can be added, with one assigned as the main one, and any of them can be used to login.

**Comment by dethe** [comment]
I think this can all be edited via the profile page, no need for a separate "settings" page.

Username is out, so just Name, and any punctuation they need for their name is fine. Not allowing punctuation is for colonizers, there's no technical reason in the 21st century to disallow punctuation or special characters in names (aside from very specific punctuation that could be mistaken for HTML or database commands, which we will quietly strip out).

Setting language is good. If someone whose language is set to Igbo visits https://participedia.net/ (the new .net, not the old one) we can redirect them to https://participedia.net/ig/ automatically. If they go directly to a language-specific page like https://participedia.net/mo/case/11233 then we won't redirect automatically, assuming they know what they're doing, but we can add a link at the top of the page `See this page in Igbo` to change from Moldovan to Igbo.

**Comment by scottofletcher** [comment]
@jesicarson @dethe Instead of a username field, are we going to have fields for First, Middle, and Last Name? And maybe just make the first field mandatory?

**Comment by dethe** [comment]
I don't think we are interested in having their first, middle, and last names. That doesn't even begin to encompass the variety of human naming and it gets complicated fast. I just want a single field where they can put their name, as they choose to give it. We can specify a preference for full, real (whatever real is, even that can have multiple meanings) names, and we can filter out people who put in obviously fake names like "I. P. Freely" or anything else Bart Simpson would come up with, but overall it is just how the user wants to be identified on the website and we don't (and shouldn't) care.

**Comment by jesicarson** [comment]
https://psacake.com/silly.asp tee hee..

**Comment by scottofletcher** [comment]
Ok, I was just thinking from the perspective of our typical user: academics and/or practitioners, on the 'older' side
